keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ArtifactSupplier.kt

```diff
@@ -84,6 +85,14 @@ interface ArtifactSupplier<A : DeliveryArtifact, V : VersioningStrategy> : Spinn
    * the metadata available within the [PublishedArtifact] object itself.
    */
   fun getGitMetadata(artifact: PublishedArtifact, versioningStrategy: VersioningStrategy): GitMetadata? = null
+
+  /**
+   * Given a [PublishedArtifact] supported by this [ArtifactSupplier],
+   * return the [ArtifactMetadata] for the artifact, if available.
+   *
+   * This function is currently expected to make calls to CI systems.
+   */
+  fun getArtifactMetadata(artifact: PublishedArtifact): ArtifactMetadata? = null
```

Comment:
My feeling is that this should probably replace `getGitMetadata` and `getBuildMetadata` on this interface... In case what prevented you from doing that was the passing of `versioningStrategy` as parameter above, the only reason I did that was because we were extracting the metadata from the artifact version string, but once we expect artifacts to carry better metadata, we could probably remove that parameter.

Reply:
P.S. Given the expectation that the call will call other services, it should probably be a `suspend` function.

Reply:
Update: after reading more of the PR I understand why you kept the other 2 functions. I suggested new names in another comment.

Reply:
Done - added suspend to the `getArtifactMetadata` + rename the build/git functions to `getDefaultXXXmetadata`